benchmarks/imagenet/resnet50/tico.py

```diff
@@ -0,0 +1,142 @@
+        projections = self.projection_head_momentum(features)
+        projections = projections.detach()
+        return projections
+
+    def training_step(
+        self, batch: Tuple[List[Tensor], Tensor, List[str]], batch_idx: int
+    ) -> Tensor:
+        momentum = cosine_schedule(
+            step=self.trainer.global_step,
+            max_steps=self.trainer.estimated_stepping_batches,
+            start_value=0.99,
+            end_value=1.0,
+        )
+        update_momentum(self.backbone, self.backbone_momentum, m=momentum)
+        update_momentum(self.projection_head, self.projection_head_momentum, m=momentum)
+
+        # Forward pass and loss calculation.
+        views, targets = batch[0], batch[1]
+
+        features = self.forward(views[0]).flatten(start_dim=1)
+        projections = self.projection_head(features)
+
+        # momentum encoder network
+        projections_momentum = self.forward_momentum(views[1])
+
+        loss = self.criterion(projections, projections_momentum)
+
+        self.log(
+            "train_loss", loss, prog_bar=True, sync_dist=True, batch_size=len(targets)
+        )
+
+        # Online linear evaluation.
+        cls_loss, cls_log = self.online_classifier.training_step(
+            (features.detach(), targets), batch_idx
+        )
+        self.log_dict(cls_log, sync_dist=True, batch_size=len(targets))
+        return loss + cls_loss
+
+    def validation_step(
+        self, batch: Tuple[Tensor, Tensor, List[str]], batch_idx: int
+    ) -> Tensor:
+        images, targets = batch[0], batch[1]
+        features = self.forward(images).flatten(start_dim=1)
+        cls_loss, cls_log = self.online_classifier.validation_step(
+            (features.detach(), targets), batch_idx
+        )
+        self.log_dict(cls_log, prog_bar=True, sync_dist=True, batch_size=len(targets))
+        return cls_loss
+
+    def configure_optimizers(self):
+        # Don't use weight decay for batch norm, bias parameters, and classification
+        # head to improve performance.
+        params, params_no_weight_decay = get_weight_decay_parameters(
+            [
+                self.backbone,
+                self.projection_head,
+            ]
+        )
+        optimizer = LARS(
+            [
+                {"name": "tico", "params": params},
+                {
+                    "name": "tico_no_weight_decay",
+                    "params": params_no_weight_decay,
+                    "weight_decay": 0.0,
+                },
+                {
+                    "name": "online_classifier",
+                    "params": self.online_classifier.parameters(),
+                    "weight_decay": 0.0,
+                },
+            ],
+            # Settings follow original code for 100 epochs which are slightly different
+            # from the paper, see:
+            # https://github.com/deepmind/deepmind-research/blob/f5de0ede8430809180254ee957abf36ed62579ef/byol/configs/byol.py#L21-L23
```

Comment:
This comment does actually not apply to TiCo. Although we could try with the BYOL settings to check whether the accuracy increases.